.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,29 @@
+name: build-and-test
```

Comment:
Separate note:  Should we enable GH for geomesa-tutorials?  I know that repo hasn't seen a lot of action recently, but a nightly build there would be shiny as early warning.

Reply:
https://geomesa.atlassian.net/browse/GEOMESA-3029 will cover this question.  :shipit: -ing it!